mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -149,15 +149,34 @@ class LogisticRegressionSuite
     assert(copiedModel.hasSummary)
   }
 
-  test("empty probabilityCol") {
-    val lr = new LogisticRegression().setProbabilityCol("")
-    val model = lr.fit(smallBinaryDataset)
+  test("empty probabilityCol or predictionCol") {
+    var lr = new LogisticRegression().setProbabilityCol("")
```

Comment:
I think we can get rid of the vars and shorten this up. Also, we should probably test both binary and multiclass. How does this look?:

````scala
val lr = new LogisticRegression().setMaxIter(1)
    val datasetFieldNames = smallBinaryDataset.schema.fieldNames.toSet
    def checkSummarySchema(model: LogisticRegressionModel, columns: Seq[String]): Unit = {
      val fieldNames = model.summary.predictions.schema.fieldNames
      assert(model.hasSummary)
      assert(datasetFieldNames.subsetOf(fieldNames.toSet))
      columns.foreach { c => assert(fieldNames.exists(_.startsWith(c))) }
    }
    Seq("binomial", "multinomial").foreach { family =>
      lr.setProbabilityCol("")
      val modelNoProb = lr.fit(smallBinaryDataset)
      checkSummarySchema(modelNoProb, Seq("probability_"))

      lr.setProbabilityCol("probability").setPredictionCol("")
      val modelNoPred = lr.fit(smallBinaryDataset)
      checkSummarySchema(modelNoPred, Seq("prediction_"))

      lr.setProbabilityCol("").setPredictionCol("")
      val modelNoPredNoProb = lr.fit(smallBinaryDataset)
      checkSummarySchema(modelNoPredNoProb, Seq("prediction_", "probability_"))
    }
````